src/util/sandbox.h

```diff
@@ -53,8 +53,14 @@ class Sandbox {
         return createSecurityToken(dir.canonicalPath(), true);
     }
 
-    static SecurityTokenPointer openSecurityToken(mixxx::FileInfo* pFileInfo, bool create);
-    static SecurityTokenPointer openSecurityTokenForDir(const QDir& dir, bool create);
+    [[nodiscard(
+            "A new security token should be assigned to a variable, otherwise "
+            "it will be invalidated immediately.")]] static SecurityTokenPointer
+    openSecurityToken(mixxx::FileInfo* pFileInfo, bool create);
+    [[nodiscard(
+            "A new security token should be assigned to a variable, otherwise "
+            "it will be invalidated immediately.")]] static SecurityTokenPointer
+    openSecurityTokenForDir(const QDir& dir, bool create);
```

Comment:
This fails unfortunately on Ubuntu Focal  
```
src/util/sandbox.h:59:62: error: wrong number of arguments specified for ‘nodiscard’ attribute
   59 |     openSecurityToken(mixxx::FileInfo* pFileInfo, bool create);
      |   
```

This works: 

```suggestion
#if defined(__GNUC__) && __cplusplus < 201907L
#define RATIONAL
#else
#define RATIONAL                                                            \
    ("A new security token should be assigned to a variable, otherwise it " \
     "will be invalidated immediately.")
#endif

    [[nodiscard RATIONAL]] static SecurityTokenPointer openSecurityToken(
            mixxx::FileInfo* pFileInfo, bool create);
    [[nodiscard RATIONAL]] static SecurityTokenPointer openSecurityTokenForDir(
            const QDir& dir, bool create);
```

Reply:
I don't think we should add workarounds for such old C++ language versions.
This is in line with our [Minimum requirements policy](https://github.com/mixxxdj/mixxx/wiki/Minimum-requirements-policy). The oldest supported Ubuntu version is currently 22.04.

Reply:
This is no reason to drop Ubuntu Focal support on the last minute before the 2.4 release. 
The alternative would be to make the rational a comment. It will anyway never appear, because the bug is now fixed.

Reply:
Than we should at least ensure, that this workaround get not merged into Main.

Reply:
I will take care.

Reply:
Nit: It should be "rationale"

Reply:
> Than we should at least ensure, that this workaround get not merged into Main.

We could simply add a commit on `main` removing this workaround after the next 2.4 -> `main` merge. That should make sure that everything not touching those lines would merge cleanly in the future.

Reply:
I have updated the suggestion.